plugin/src/HeadlessAPI.lua

```diff
@@ -0,0 +1,452 @@
+		assert(type(port) == "string" or port == nil, "Port must be type `string?`")
+
+		if Rojo:_checkRateLimit("ConnectAsync") then
+			return
+		end
+
+		app:startSession(host, port)
+	end
+
+	Rojo._apiDescriptions.DisconnectAsync = "Disconnects from the Rojo server"
+	function Rojo:DisconnectAsync()
+		if Rojo:_checkRateLimit("DisconnectAsync") then
+			return
+		end
+
+		app:endSession()
+	end
+
+	Rojo._apiDescriptions.GetSetting = "Gets a Rojo setting"
+	function Rojo:GetSetting(setting: string): any
+		assert(type(setting) == "string", "Setting must be type `string`")
+
+		return Settings:get(setting)
+	end
+
+	Rojo._apiDescriptions.SetSetting = "Sets a Rojo setting"
+	function Rojo:SetSetting(setting: string, value: any)
+		assert(type(setting) == "string", "Setting must be type `string`")
+
+		if Rojo:_checkRateLimit("SetSetting") then
+			return
+		end
+
+		return Settings:set(setting, value)
+	end
+
+	Rojo._apiDescriptions.Notify = "Shows a notification in the Rojo UI"
+	function Rojo:Notify(
+		msg: string,
+		timeout: number?,
+		actions: { [string]: { text: string, style: string, layoutOrder: number, onClick: (any) -> () } }?
+	): () -> ()
+		assert(type(msg) == "string", "Message must be type `string`")
+		assert(type(timeout) == "number" or timeout == nil, "Timeout must be type `number?`")
+		assert((actions == nil) or (type(actions) == "table"), "Actions must be table or nil")
+
+		if Rojo:_checkRateLimit("Notify") then
+			return function() end
+		end
+
+		local sanitizedActions = nil
+		if actions then
+			sanitizedActions = {}
+			for id, action in actions do
+				assert(type(id) == "string", "Actions key must be string")
+				local actionId = "Actions." .. id
+				assert(type(action) == "table", actionId .. " must be table")
+				assert(type(action.text) == "string", actionId .. ".text must be string")
+				assert(type(action.style) == "string", actionId .. ".style must be string")
+				assert(
+					action.style == "Solid" or action.style == "Bordered",
+					actionId .. ".style must be 'Solid' or 'Bordered'"
+				)
+				assert(type(action.layoutOrder) == "number", actionId .. ".layoutOrder must be number")
+				assert(type(action.onClick) == "function", actionId .. ".onClick must be function")
+
+				sanitizedActions[id] = {
+					text = action.text,
+					style = action.style,
+					layoutOrder = action.layoutOrder,
+					onClick = action.onClick,
+				}
+			end
+		end
+
+		return app:addThirdPartyNotification(Rojo:_getCallerName(), msg, timeout, sanitizedActions)
+	end
+
+	Rojo._apiDescriptions.GetHostAndPort = "Gets the host and port that Rojo is set to"
+	function Rojo:GetHostAndPort(): (string, string)
+		return app:getHostAndPort()
+	end
+
+	Rojo._apiDescriptions.CreateApiContext = "Creates a new API context"
+	function Rojo:CreateApiContext(baseUrl: string)
+		assert(type(baseUrl) == "string", "Base URL must be type `string`")
+
+		return ApiContext.new(baseUrl)
+	end
```

Comment:
Is there a usecase for creating an `ApiContext`? Users wouldn't know what `CreateApiContext` is for.

Reply:
I added it because I needed it for Rojo Scanner, so yeah there's use cases. ApiContext lets you interact with Rojo server endpoints. Without being able to create an ApiContext through this, you'd need to reimplement and maintain Rojo endpoint usage in your plugins (which would suck). It's really only needed for more complex uses.